### PR TITLE
D3 modules

### LINKED ISSUE #12538: d3-box and d3-slider need to be upgraded for types-2.0
- [X] I want to talk about `d3-box` and `d3-slider`.
  - The authors of that type definition are cc/ @lk-chen

These two packages are currently lacking a `tsconfig.json`. They seem to be trying to augment the `d3` namespace, but `d3` is now a module. What should be done with these?

CC @tomwanzek

### PR DESCRIPTION
**Pull request by ghost**
- [X] Prefer to make your PR against the `types-2.0` branch.
- [ ] Test the change in your own code.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).

If changing an existing definition:
- [ ] Provide a URL to  documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.

Fixes #12538. These packages had grown out-of-date.
I've upgraded them to have a `package.json` that specifies using d3 version 3 typings, and to use module augmentation rather than namespace augmentation.
Thanks @tomwanzek for the suggestion.

### CODE REVIEW

**Comment by tomwanzek** [comment]
@andy-ms No worries. Let me know, if something does not work as expected,

File: d3-box/index.d.ts [hunk 1]
```diff
@@ -3,9 +3,9 @@
 // Definitions by: Linkun Chen <https://github.com/lk-chen>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
-/// <reference path="../d3/d3.d.ts"/>
```
Comment (vvakame) [inline d3-box/index.d.ts]: We need `/// <reference types="d3" />`.

Comment (tomwanzek) [inline d3-box/index.d.ts]: 👍 Yes, that's the pattern we used for the old definitions, due to the way they were written.

Comment (andy-ms) [inline d3-box/index.d.ts]: The code compiles without an explicit reference -- that's pulled in by the import.

**Review by vvakame — APPROVED** [review]

**Pull request merged**